docs/src/userguide/language_basics.rst

```diff
@@ -381,6 +381,48 @@ Sometimes Cython will complain unnecessarily, and sometimes it will fail to
 detect a problem that exists. Ultimately, you need to understand the issue and
 be careful what you do.
 
+
+Type Casting
+------------
+
+As C uses the ``"("`` and ``")"`` for casting types, Cython uses ``"<"`` and ``">"``
+to signify a casting. For example::
+
+    cdef char *p
+    cdef float *q
+    p = <char*>q
+
+If one of the types is a Python object for ``<type>x``, Cython will try to do a coercion.
+
+ .. note:: Cython will not stop a casting where there is no conversion, but it will emit a warning.
+
+To get the address of some Python object, use a cast to a pointer type
+like ``<void*>`` or ``<PyObject*>``.
+You can also get the python object from a pointer to this object by doing ``<object>``.
```

Comment:
"You can also cast a C pointer back to a Python object reference with `<object>`, or a more specific builtin or extension type (e.g. `<MyExtType>ptr`). This will increase the reference count of the object by one, i.e. the cast returns an owned reference."